infra/main.bicep

```diff
@@ -0,0 +1,703 @@
+  name: 'search-role-function'
+  params: {
+    principalId: function.outputs.FUNCTION_IDENTITY_PRINCIPAL_ID
+    roleDefinitionId: '8ebe5a00-799e-43f5-93ac-243d3dce84a7'
+    principalType: 'ServicePrincipal'
+  }
+}
+
+output APPLICATIONINSIGHTS_CONNECTION_STRING string = monitoring.outputs.applicationInsightsConnectionString
+output APPINSIGHTS_INSTRUMENTATIONKEY string = monitoring.outputs.applicationInsightsInstrumentationKey
+output AZURE_KEY_VAULT_ENDPOINT string = useKeyVault ? keyvault.outputs.endpoint : ''
+output AZURE_KEY_VAULT_NAME string = useKeyVault || authType == 'rbac' ? keyvault.outputs.name : ''
+output AZURE_LOCATION string = location
+output AZURE_TENANT_ID string = tenant().tenantId
+output AZURE_CONTENT_SAFETY_ENDPOINT string = contentsafety.outputs.endpoint
+output AZURE_SEARCH_SERVICE string = search.outputs.endpoint
+output AZURE_FORM_RECOGNIZER_ENDPOINT string = formrecognizer.outputs.endpoint
+output AZURE_SEARCH_USE_SEMANTIC_SEARCH string = azureSearchUseSemanticSearch
+output AZURE_SEARCH_SEMANTIC_SEARCH_CONFIG string = azureSearchSemanticSearchConfig
+output AZURE_SEARCH_INDEX_IS_PRECHUNKED string = azureSearchIndexIsPrechunked
+output AZURE_SEARCH_TOP_K string = azureSearchTopK
+output AZURE_SEARCH_ENABLE_IN_DOMAIN string = azureSearchEnableInDomain
+output AZURE_SEARCH_CONTENT_COLUMNS string = azureSearchContentColumns
+output AZURE_SEARCH_FILENAME_COLUMN string = azureSearchFilenameColumn
+output AZURE_SEARCH_TITLE_COLUMN string = azureSearchTitleColumn
+output AZURE_SEARCH_URL_COLUMN string = azureSearchUrlColumn
+output AZURE_OPENAI_MODEL_NAME string = azureOpenAIModelName
+output AZURE_OPENAI_STREAM string = azureOpenAIStream
+output AZURE_OPENAI_SYSTEM_MESSAGE string = azureOpenAISystemMessage
+output AZURE_OPENAI_STOP_SEQUENCE string = azureOpenAIStopSequence
+output AZURE_OPENAI_MAX_TOKENS string = azureOpenAIMaxTokens
+output AZURE_OPENAI_TOP_P string = azureOpenAITopP
+output AZURE_OPENAI_TEMPERATURE string = azureOpenAITemperature
+output AZURE_SEARCH_INDEX string = azureSearchIndex
+output AZURE_OPENAI_API_VERSION string = azureOpenAIApiVersion
+output DOCUMENT_PROCESSING_QUEUE_NAME string = queueName
+output AZURE_BLOB_CONTAINER_NAME string = blobContainerName
+output AZURE_BLOB_ACCOUNT_NAME string = storageAccountName
+output AZURE_OPENAI_RESOURCE string = azureOpenAIResourceName
+output AZURE_OPENAI_EMBEDDING_MODEL string = azureOpenAIEmbeddingModel
+output AZURE_OPENAI_MODEL string = azureOpenAIModel
+output USE_KEY_VAULT bool = useKeyVault
+output AZURE_OPENAI_KEY string = useKeyVault ? storekeys.outputs.OPENAI_KEY : ''
+output AZURE_BLOB_ACCOUNT_KEY string = useKeyVault ? storekeys.outputs.STORAGE_ACCOUNT_KEY: ''
+output AZURE_FORM_RECOGNIZER_KEY string = useKeyVault ? storekeys.outputs.FORM_RECOGNIZER_KEY: ''
+output AZURE_SEARCH_KEY string = useKeyVault ? storekeys.outputs.SEARCH_KEY: ''
+output AZURE_CONTENT_SAFETY_KEY string = useKeyVault ? storekeys.outputs.CONTENT_SAFETY_KEY: ''
```

Comment:
You cannot output keys in bicep

Reply:
The keys here are the key names stored in the `KeyVault`, not the actual key values.

Reply:
Please update the output to *KEY_NAME